### PR TITLE
Filter null values before plotting

### PR DESCRIPTION
**Pull request by Illviljan**
I noticed that seaborn's plot was responding much faster than xarray's version with the same data. 
Turn's out seaborn drops any nulls: https://github.com/mwaskom/seaborn/blob/056413d7393e3daec597d430c076e45938d53376/seaborn/relational.py#L399

- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

**Review by max-sixty — APPROVED** [review]
Good idea!

**Comment by headtr1ck** [comment]
Isn't matplotlib dropping nans anyway? I never had any issues with nan values in my data...

**Comment by Illviljan** [comment]
No, they aren't dropped. That's what the test shows. Do you use an interactive version when you look at plots? It was painfully slow when I was zooming in and out.

I was able to reproduce the problem with a raw matplotlib version as well, removing the nans was indeed the solution.

**Comment by headtr1ck** [comment]
> No, they aren't dropped. That's what the test shows. Do you use an interactive version when you look at plots? It was painfully slow when I was zooming in and out.
> 
> I was able to reproduce the problem with a raw matplotlib version as well, removing the nans was indeed the solution.

ok, never look at the plots interactively...
Then its fine to merge this :)

**Review by headtr1ck — APPROVED** [review]

File: xarray/plot/dataarray_plot.py [hunk 1]
```diff
@@ -948,6 +948,12 @@ def newplotfunc(
             size_ = kwargs.pop("_size", linewidth)
             size_r = _LINEWIDTH_RANGE
 
+        # Remove any nulls, .where(m, drop=True) doesn't work when m is a dask array,
```
Comment (dcherian) [inline xarray/plot/dataarray_plot.py:951]: This can only be done for scatter plots. A line plot would totally change.

Comment (Illviljan) [reply xarray/plot/dataarray_plot.py:951]: Currently only scatter uses `plot1d` so it should be fine for now.
Seaborn doesn't seem to filter nan on line however.
Can't move it later in the chain as this nan-trick is used to split on different dimensions, can move it to the if check above though

NaN's are usually usually not wanted in line plots either in my experience. Do you have a usecase in mind?

Comment (Illviljan) [reply xarray/plot/dataarray_plot.py]: Moved the null filter inside a scatter check.

Comment (dcherian) [reply xarray/plot/dataarray_plot.py]: > NaN's are usually not wanted in line plots either in my experience. Do you have a usecase in mind?

Plenty. it's important to know where the gaps are. That's useful information. It is annoying that matplotlib will skip NaNs at the beginning and ends of a line plot.

**Pull request merged**